### PR TITLE
#64 Doubled prefix operators "!!" and "~~" should not be used

### LINKED ISSUE #64: Rule: Double negation operator
Replaces http://jira.sonarsource.com/browse/LINTVS-172

Using `!!` doesn't make sense, and can be safely removed.

Exception: if the negation operator is overloaded, then we shouldn't report on it. 

R# ID: *DoubleNegationOperator*

Implements: http://jira.sonarsource.com/browse/RSPEC-2761

### PR DESCRIPTION
**Pull request by tvsonar**
Fixes #64

### CODE REVIEW

**Comment by tvsonar** [comment]
The rule description needs to be updated.

File: src/SonarLint/SonarLint.csproj [hunk 1]
```diff
@@ -405,7 +409,7 @@
     </ProjectReference>
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
+  <!-- To modify your build process, add your task inside one of the targets below and uncomment it.
```
Comment (dbolkensteyn) [inline src/SonarLint/SonarLint.csproj]: @tvsonar this unrelated change should not go into this PR

File: src/SonarLint/Rules/UnaryPrefixOperatorRepeated.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+using SonarLint.Common;
+using SonarLint.Common.Sqale;
+using SonarLint.Helpers;
+using Microsoft.CodeAnalysis.Text;
+
+namespace SonarLint.Rules
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [SqaleConstantRemediation("5min")]
+    [SqaleSubCharacteristic(SqaleSubCharacteristic.Understandability)]
```
Comment (dbolkensteyn) [inline src/SonarLint/Rules/UnaryPrefixOperatorRepeated.cs]: @tvsonar SQALE has changed

File: src/SonarLint/Rules.Description/S2761.html [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<p>
+    Calling the <code>!</code> or <code>~</code> prefix operator twice does nothing: the second invocation undoes the first.
```
Comment (dbolkensteyn) [inline src/SonarLint/Rules.Description/S2761.html]: @tvsonar the description has changed

**Comment by dbolkensteyn** [comment]
LGTM
